### PR TITLE
Add multiple riders support

### PR DESCRIPTION
**Pull request by d3m0**
Implemented multiple riders on track.
Added ability to set rider number by default or throw dialog if default number is empty.

### CODE REVIEW

File: src/main/java/com/d3m0/ts/core/Rider.java [hunk 1]
```diff
@@ -32,11 +28,11 @@ public void setRiderEndTime(long riderEndTime) {
...
-    public void start() {
-        riderStartTime = System.currentTimeMillis();
+    public long start() {
+        return riderStartTime = System.currentTimeMillis();
     }
 
-    public void stop() {
-
+    public Double getTimeOnTrack() {
+        return (getRiderEndTime() - getRiderStartTime()) / 1000.00;
```
Comment (artem-shevchuk84) [inline src/main/java/com/d3m0/ts/core/Rider.java]: Extract into method

File: src/main/java/com/d3m0/ts/core/Race.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    public void setFinishNumber(String finishNumber) {
+        this.finishNumber = finishNumber;
+    }
+
+    public Rider startTime(Action action) {
+        action.start();
+        rider = new Rider();
+        rider.start();
+        if (getStartNumber().isEmpty()) {
+            rider.setRiderNumber(new InputNumberDialog().getNumber("Who's starting?"));
```
Comment (artem-shevchuk84) [inline src/main/java/com/d3m0/ts/core/Race.java]: Does it need validation? Whats happened if number will be wrong valued

File: src/main/java/com/d3m0/ts/core/Race.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        } else {
+            rider.setRiderNumber(getStartNumber());
+            setStartNumber("");
```
Comment (artem-shevchuk84) [inline src/main/java/com/d3m0/ts/core/Race.java]: Set as constant DEFAULT_NUMBER

File: src/main/java/com/d3m0/ts/core/Race.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        }
+        Rider finishedRider = new Rider();
+
+        for (Rider riderOnTrack : ridersOnTrack) {
+            if (riderOnTrack.getRiderNumber().equals(getFinishNumber())) {
+                riderOnTrack.setRiderEndTime(stopTime);
+                finishedRider = riderOnTrack;
+            }
+        }
+        setFinishNumber("");
```
Comment (artem-shevchuk84) [inline src/main/java/com/d3m0/ts/core/Race.java]: Set as constant DEFAULT_NUMBER

File: src/main/java/com/d3m0/ts/core/Race.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        return rider;
+    }
+
+    public Rider stopTime(Action action) {
+        action.stop();
+        long stopTime = System.currentTimeMillis();
+        if (getFinishNumber().isEmpty()) {
+            setFinishNumber(new InputNumberDialog().getNumber("Who's finished?"));
+        }
+        Rider finishedRider = new Rider();
```
Comment (artem-shevchuk84) [inline src/main/java/com/d3m0/ts/core/Race.java]: Why new object for?

File: src/main/java/com/d3m0/ts/core/Race.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+        for (Rider riderOnTrack : ridersOnTrack) {
+            if (riderOnTrack.getRiderNumber().equals(getFinishNumber())) {
+                riderOnTrack.setRiderEndTime(stopTime);
+                finishedRider = riderOnTrack;
+            }
```
Comment (artem-shevchuk84) [inline src/main/java/com/d3m0/ts/core/Race.java]: Else throw exception

File: src/main/java/com/d3m0/ts/gui/InputNumberDialog.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
+package com.d3m0.ts.gui;
+
+import javax.swing.*;
+import java.awt.event.*;
+
+public class InputNumberDialog extends JDialog {
+    private JPanel contentPane;
+    private JButton buttonOK;
+    private JButton buttonCancel;
+    private JTextField textField1;
```
Comment (artem-shevchuk84) [inline src/main/java/com/d3m0/ts/gui/InputNumberDialog.java]: Bazinga

File: pom.xml [hunk 8]
```diff
@@ -15,4 +20,4 @@
             <version>4.2.2.RELEASE</version>
```
Comment (IevgenTkachuk) [inline pom.xml]: This should be defined in properties <springframework>4.2.2.RELEASE</springframework> then use ${springframework}

Comment (artem-shevchuk84) [inline pom.xml]: +1

File: src/main/java/com/d3m0/ts/core/Race.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+    public Rider stopTime(Action action) {
+        action.stop();
+        long stopTime = System.currentTimeMillis();
+        if (getFinishNumber().isEmpty()) {
+            setFinishNumber(new InputNumberDialog().getNumber("Who's finished?"));
+        }
+        Rider finishedRider = new Rider();
+
+        for (Rider riderOnTrack : ridersOnTrack) {
```
Comment (IevgenTkachuk) [inline src/main/java/com/d3m0/ts/core/Race.java]: Use HasMap<Integer,Rider> for riderOnTrack

File: src/main/java/com/d3m0/ts/gui/InputNumberDialog.form [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+            </constraints>
+            <properties/>
+            <border type="none"/>
+            <children>
+              <component id="e7465" class="javax.swing.JButton" binding="buttonOK">
+                <constraints>
+                  <grid row="0" column="0" row-span="1" col-span="1" vsize-policy="0" hsize-policy="3" anchor="0" fill="1" indent="0" use-parent-layout="false"/>
+                </constraints>
+                <properties>
+                  <text value="OK"/>
```
Comment (IevgenTkachuk) [inline src/main/java/com/d3m0/ts/gui/InputNumberDialog.form]: Use properties file for all text values

File: src/main/java/com/d3m0/ts/core/Rider.java (hunk 1 above)
Comment (d3m0) [inline src/main/java/com/d3m0/ts/core/Rider.java]: fixed

File: src/main/java/com/d3m0/ts/core/Race.java (hunk 2 above)
Comment (d3m0) [inline src/main/java/com/d3m0/ts/core/Race.java]: Need to discuss this offline

File: src/main/java/com/d3m0/ts/core/Race.java (hunk 3 above)
Comment (d3m0) [inline src/main/java/com/d3m0/ts/core/Race.java]: Refactored code. removed unnecessary variables

File: src/main/java/com/d3m0/ts/core/Race.java (hunk 4 above)
Comment (d3m0) [inline src/main/java/com/d3m0/ts/core/Race.java]: Refactored code. removed unnecessary variables

File: src/main/java/com/d3m0/ts/core/Race.java (hunk 5 above)
Comment (d3m0) [inline src/main/java/com/d3m0/ts/core/Race.java]: Refactored code. Removed unnecessary objects

File: pom.xml (hunk 8 above)
Comment (d3m0) [inline pom.xml]: Fixed

File: src/main/java/com/d3m0/ts/core/Race.java (hunk 9 above)
Comment (d3m0) [inline src/main/java/com/d3m0/ts/core/Race.java]: Refactored code

File: src/main/java/com/d3m0/ts/gui/InputNumberDialog.form (hunk 10 above)
Comment (d3m0) [inline src/main/java/com/d3m0/ts/gui/InputNumberDialog.form]: Need to discuss offline

**Pull request merged**